### ISSUE TITLE
Can't export mesh

### ISSUE THREAD

**Issue by manavortex** [WolvenKit/WolvenKit#1162]
**Describe the bug**
Exception when trying to export mesh: 
[source_.zip](https://github.com/WolvenKit/WolvenKit/files/10758844/source_.zip)
```

[16/02/2023 17:48:28] manavortex\amm_props\pink_stuff\hair_ties_box\hair_ties_box.mesh - Object reference not set to an instance of an object.
[16/02/2023 17:48:28] Message: Object reference not set to an instance of an object.
Source: WolvenKit.RED4
StackTrace:    at WolvenKit.RED4.Archive.FileEntry.Extract(Stream output, Boolean decompressBuffers) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\Base\FileEntry.cs:line 93
   at WolvenKit.RED4.Archive.FileEntry.Extract(Stream output) in D:\a\WolvenKit\WolvenKit\WolvenKit.RED4\Archive\Base\FileEntry.cs:line 87
   at WolvenKit.Modkit.RED4.ModTools.InternalTryFindFile(List`1 archives, UInt64 hash, FindFileEntry& result, Boolean excludeCustomArchives) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\ModTools.Helper.cs:line 82
   at WolvenKit.Modkit.RED4.ModTools.TryFindFile(List`1 archives, ResourcePath path, FindFileEntry& result, Boolean excludeCustomArchives) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\ModTools.Helper.cs:line 38
   at WolvenKit.Modkit.RED4.ModTools.<ParseMaterials>g__ExtractMlSetup|51_4(String path, <>c__DisplayClass51_0&) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Tools\MaterialTools.cs:line 752
   at WolvenKit.Modkit.RED4.ModTools.<ParseMaterials>g__ExtractFile|51_0(String path, <>c__DisplayClass51_0&) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Tools\MaterialTools.cs:line 665
   at WolvenKit.Modkit.RED4.ModTools.ParseMaterials(CR2WFile cr2w, Stream meshStream, FileInfo outfile, List`1 archives, String matRepo, MeshesInfo info, EUncookExtension eUncookExtension) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Tools\MaterialTools.cs:line 610
   at WolvenKit.Modkit.RED4.ModTools.ExportMesh(Stream meshStream, FileInfo outfile, MeshExportArgs meshExportArgs, ValidationMode vmode) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Uncook.cs:line 853
   at WolvenKit.Modkit.RED4.ModTools.HandleMesh(Stream cr2wStream, FileInfo cr2wFileName, MeshExportArgs meshargs) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Uncook.cs:line 864
   at WolvenKit.Modkit.RED4.ModTools.UncookBuffers(Stream cr2wStream, String relPath, GlobalExportArgs settings, DirectoryInfo rawOutDir, ECookedFileFormat[] forcebuffers) in D:\a\WolvenKit\WolvenKit\WolvenKit.Modkit\RED4\Uncook.cs:line 432
[16/02/2023 17:48:29] The following items failed:
E:\CyberpunkFiles\AMM_stuff\AMM_pink_props\source\archive\manavortex\amm_props\pink_stuff\hair_ties_box\hair_ties_box.mesh
```

This is something of a heisenbug. Maybe importing/exporting the mesh with a previous WKit version fixed it, or maybe reopening WKit made it go away.

I duplicated one of the other folders (although I don't remember which one), then renamed both the folder and the files inside. I _think_ i had WKit open during that time.
Exporting the newly-renamed mesh didn't work, exporting all the originals did. Maybe it has something to do with caching, e.g. it trying to match the file checksum or whatever and not liking that it changed? I have no idea.

Filing the bug anyway, in case it's of some help. 

**Version:**
8.8.2-nightly.2023-02-16

**Comment by dragonzkiller** [comment]
This is due to this change: https://github.com/WolvenKit/WolvenKit/blame/main/WolvenKit.App/Models/FileSystemArchive.cs#L63

Which ultimately causes the error here:
https://github.com/WolvenKit/WolvenKit/blob/main/WolvenKit.RED4/Archive/Base/FileEntry.cs#L93

Basically `Archive` is `null` due to the change and it won't pass the check, then fails when it even tries to throw the exception because... it's `null`.

Not sure why it was done this way, you'll have to ask fuzzo. I'll play around a bit and see what I can do.

**Comment by seberoth** [comment]
Fixed with #1170
